dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ZenodoHelper.java

```diff
@@ -299,18 +300,29 @@ private static void setMetadataRelatedIdentifiers(DepositMetadata depositMetadat
      * @param depositMetadata Metadata for the workflow version
      * @param workflow    workflow for which DOI is registered
      */
-    static void setMetadataCreator(DepositMetadata depositMetadata, Workflow workflow) {
-        final Stream<Author> authors = workflow.getAuthors().stream().map(ZenodoHelper::fromDockstoreAuthor);
-        final Stream<Author> orcidAuthors = workflow.getOrcidAuthors().stream()
+    static void setMetadataCreator(DepositMetadata depositMetadata, Workflow workflow, WorkflowVersion workflowVersion) {
+        // prefer authors from the specific workflow version
+        final Set<Author> setOfAuthors = new HashSet<>(getAuthors(workflowVersion.getAuthors(), workflowVersion.getOrcidAuthors()));
+        /// but use the default if necessary
+        if (setOfAuthors.isEmpty()) {
+            final List<Author> zenodoAuthors = getAuthors(workflow.getAuthors(), workflow.getOrcidAuthors());
+            setOfAuthors.addAll(zenodoAuthors);
```

Comment:
Does the Zenodo `Author` class define `equals` and `hashCode` appropriately so that when we add equivalent instances to the `Set`, the `Set` can tell, and de-duplicate them?  We probably should check to make sure.

Reply:
They looked sane, but always good to double-check. Added test